src/System.Device.Gpio.Tests/GpioControllerTestBase.cs

```diff
@@ -153,7 +150,75 @@ void callback(object sender, PinValueChangedEventArgs pinValueChangedEventArgs)
             }
         }
 
+        [Fact]
+        public void AddCallbackRemoveCallbackTest()
+        {
+            int risingEventOccuredCount = 0, fallingEventOccuredCount = 0;
+            using (GpioController controller = new GpioController(GetTestNumberingScheme(), GetTestDriver()))
+            {
+                controller.OpenPin(InputPin, PinMode.Input);
+                controller.OpenPin(OutputPin, PinMode.Output);
+
+                controller.RegisterCallbackForPinValueChangedEvent(InputPin, PinEventTypes.Rising, (o, e) => {
+                    risingEventOccuredCount++;
+                });
+                PinChangeEventHandler callback = (o, e) => {
+                    risingEventOccuredCount++;
+                };
+                controller.RegisterCallbackForPinValueChangedEvent(InputPin, PinEventTypes.Rising, callback);
+                controller.RegisterCallbackForPinValueChangedEvent(InputPin, PinEventTypes.Rising, (o, e) => {
+                    risingEventOccuredCount++;
+                    if (fallingEventOccuredCount == 4)
+                    {
+                        controller.UnregisterCallbackForPinValueChangedEvent(InputPin, callback);
+                    }
+                });
+
+                controller.RegisterCallbackForPinValueChangedEvent(InputPin, PinEventTypes.Falling, (o, e) => {
+                    fallingEventOccuredCount++;
+                });
+                for (int i = 0; i < 10; i++)
+                {
+                    controller.Write(OutputPin, PinValue.High);
+                    Thread.Sleep(100);
```

Comment:
why is sleep needed?

Reply:
Sleep needed to make sure hardware able to detect the event, about the large number -> was not really sure how much sleep is enough, so just put enough duration 😛

Reply:
set to sleep 10 milliseconds